catroid/src/main/java/org/catrobat/catroid/devices/arduino/ArduinoListener.java

```diff
@@ -164,11 +155,28 @@ public int getAnalogPin5() {
 		return analogPin5;
 	}
 
-	public int getPortValue(int pin) {
-		return portValue[pin];
+	public int getPinValue(int pin) {
+		if ((pin >= 0) && (pin < ArduinoImpl.NUMBER_OF_DIGITAL_PINS)) {
+			int port = (pin + 7) / 8;
+			int index = pin % 8;
+			return ArduinoImpl.getBit(this.portValue[port], index);
+		}
+		return 0;
+	}
+
+	public void setPinValue(int pin, int value) {
+		if ((pin >= 0) && (pin < ArduinoImpl.NUMBER_OF_DIGITAL_PINS)) {
+			int port = (pin + 7) / 8;
+			int index = pin % 8;
+			this.portValue[port] = ArduinoImpl.setBit(this.portValue[port], index, value);
+		}
+	}
+
+	public int getPortValue(int port) {
+		return portValue[port];
 	}
 
-	public void setPortValue(int pin, int value) {
-		this.portValue[pin] = value;
+	public void setPortValue(int port, int value) {
+		this.portValue[port] = value;
```

Comment:
Remove setter. The port pin values are either set through `setPinValue` or updated in `onDigitalMessageReceived`.